### PR TITLE
Indexer API query parameters changes and checks

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -230,6 +230,9 @@ pub enum Error {
     /// Local time doesn't match the time of the latest milestone timestamp
     #[error("Local time {0} doesn't match the time of the latest milestone timestamp: {1}")]
     TimeNotSynced(u32, u32),
+    /// An indexer API request contains a query parameter not supported by the endpoint.
+    #[error("An indexer API request contains a query parameter not supported by the endpoint.")]
+    UnsupportedQueryParameter,
```
Comment (Thoralf-M) [inline src/error.rs]: Wouldn't it be better to also return which one is wrong?

Comment (thibault-martinez) [reply src/error.rs]: Done

File: src/error.rs [hunk 2]
```diff
@@ -232,7 +234,7 @@ pub enum Error {
     TimeNotSynced(u32, u32),
     /// An indexer API request contains a query parameter not supported by the endpoint.
     #[error("An indexer API request contains a query parameter not supported by the endpoint.")]
-    UnsupportedQueryParameter,
+    UnsupportedQueryParameter(QueryParameter),
```
Comment (Thoralf-M) [inline src/error.rs:237]: I think for thiserror it should also be added in the message above with `{0}`

Comment (thibault-martinez) [reply src/error.rs:237]: Ok but it means I have to impl Display for QueryParameter, are we fine with the impl just calling `to_query_string` ?

Comment (thibault-martinez) [reply src/error.rs:237]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**